test/built-ins/Function/prototype/toString/intrinsics.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2016 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-function.prototype.tostring
+description: Function.prototype.toString on well-known intrinsic function objects
+---*/
+
+let intrinsics = {
+  Array, ArrayBuffer, Boolean, DataView, Date, decodeURI, decodeURIComponent, encodeURI,
+  encodeURIComponent, Error, eval, EvalError, Float32Array, Float64Array, Function, Int8Array,
+  Int16Array, Int32Array, isFinite, isNaN, Map, Number, Object, parseFloat, parseInt, Promise,
+  Proxy, RangeError, ReferenceError, RegExp, Set, String, Symbol, SyntaxError, TypeError,
+  Uint8Array, Uint8ClampedArray, Uint16Array, Uint32Array, URIError, WeakMap, WeakSet,
+};
+
+const NATIVE_FUNCTION_RE = /\bfunction\b[\s\S]+\b\w+\b[\s\S]*\([\s\S]*\)[\s\S]*\{[\s\S]*\[[\s\S]*\bnative\b[\s\S]+\bcode\b[\s\S]*\][\s\S]*\}/;
```

Comment:
This regular looks too permissive for the syntax defined by your proposal:

> NativeFunction :
>     function BindingIdentifieropt ( FormalParameters ) { [ native code ] }

For funsies, I wrote a function that satisfies the RegExp but violates the
spec:

    function eval ( x = 'function() { [ native code ] }' ) { 'lol'; }

It seems like:

- most of the `\S` classes may be removed (the one in the FormalParameters
  position seems necessary)
- the pattern should be anchored with `^\s*` and `\s*$`
- the `\b` classes may be removed